src/test/java/uk/gov/pay/api/service/GetPaymentEventServiceTest.java

```diff
@@ -0,0 +1,69 @@
+import org.mockito.junit.MockitoJUnitRunner;
+import uk.gov.pay.api.app.RestClientFactory;
+import uk.gov.pay.api.app.config.PublicApiConfig;
+import uk.gov.pay.api.app.config.RestClientConfig;
+import uk.gov.pay.api.auth.Account;
+import uk.gov.pay.api.model.PaymentEventsResponse;
+import uk.gov.pay.api.model.TokenPaymentType;
+import uk.gov.pay.commons.testing.pact.consumers.PactProviderRule;
+import uk.gov.pay.commons.testing.pact.consumers.Pacts;
+
+import javax.ws.rs.client.Client;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class GetPaymentEventServiceTest {
+
+    @Rule
+    public PactProviderRule connectorRule = new PactProviderRule("connector", this);
+
+    @Mock
+    private PublicApiConfig mockConfiguration;
+
+    private GetPaymentEventService getPaymentEventService;
+    private static final String ACCOUNT_ID = "42";
+
+    @Before
+    public void setUp() throws Exception {
+        when(mockConfiguration.getConnectorUrl()).thenReturn(connectorRule.getUrl());
+        when(mockConfiguration.getBaseUrl()).thenReturn("http://publicapi.test.localhost/");
+
+        PublicApiUriGenerator publicApiUriGenerator = new PublicApiUriGenerator(mockConfiguration);
+        ConnectorUriGenerator connectorUriGenerator = new ConnectorUriGenerator(mockConfiguration);
+        Client client = RestClientFactory.buildClient(new RestClientConfig(false));
+
+        getPaymentEventService = new GetPaymentEventService(client, connectorUriGenerator, publicApiUriGenerator);
+    }
+
+    @Test
+    @PactVerification("connector")
+    @Pacts(pacts = {"publicapi-connector-get-payment-events"})
+    public void shouldReturnPaymentEvents() {
+        Account account = new Account(ACCOUNT_ID, TokenPaymentType.CARD);
+        PaymentEventsResponse paymentEventsResponse = getPaymentEventService.getPaymentEvent(account, "abc123");
+        assertThat(paymentEventsResponse.getPaymentId(), is("abc123"));
+        assertThat(paymentEventsResponse.getLinks().getSelf().getHref(), is("http://publicapi.test.localhost/v1/payments/abc123/events"));
+        assertThat(paymentEventsResponse.getEvents().size(), is(2));
+        assertThat(paymentEventsResponse.getEvents().get(0).getPaymentId(), is("abc123"));
+        assertThat(paymentEventsResponse.getEvents().get(0).getPaymentLink().getPaymentLink().getHref(), is("http://publicapi.test.localhost/v1/payments/abc123"));
+        assertThat(paymentEventsResponse.getEvents().get(0).getState().getStatus(), is("created"));
+        assertThat(paymentEventsResponse.getEvents().get(0).getState().isFinished(), is(false));
+        assertThat(paymentEventsResponse.getEvents().get(1).getPaymentId(), is("abc123"));
+        assertThat(paymentEventsResponse.getEvents().get(1).getPaymentLink().getPaymentLink().getHref(), is("http://publicapi.test.localhost/v1/payments/abc123"));
```

Comment:
would be good to check whether `updated` is populated as well

Reply:
addressed in https://github.com/alphagov/pay-publicapi/pull/633/commits/14311e18b8df8018761b8be7218dbe73e6332284